cmd/osm-controller/validate_test.go

```diff
@@ -1,160 +1,149 @@
-
-		})
-	})
-	Context("cert-manager certProviderKind is passed in with valid caBundleSecretName and certmanagerIssuerName", func() {
-		certProviderKind = providers.CertManagerKind.String()
-		caBundleSecretName = testCaBundleSecretName
-		certManagerOptions.IssuerName = "test-issuer"
-
-		err := validateCertificateManagerOptions()
-
-		It("should not error", func() {
-			Expect(err).To(BeNil())
-		})
-	})
-	Context("cert-manager certProviderKind is passed in with caBundleSecretName but no certmanagerIssureName", func() {
-		certProviderKind = providers.CertManagerKind.String()
-		caBundleSecretName = testCaBundleSecretName
-		certManagerOptions.IssuerName = ""
-
-		err := validateCertificateManagerOptions()
-
-		It("should error", func() {
-			Expect(err).To(HaveOccurred())
-		})
-	})
-	Context("cert-manager certProviderKind is passed in without caBundleSecretName but no certmanagerIssureName", func() {
-		certProviderKind = providers.CertManagerKind.String()
-		caBundleSecretName = ""
-		certManagerOptions.IssuerName = ""
-
-		err := validateCertificateManagerOptions()
-
-		It("should error", func() {
-			Expect(err).To(HaveOccurred())
+func TestValidateCertificateManagerOptions(t *testing.T) {
+	assert := tassert.New(t)
+
+	testCases := []struct {
+		name               string
+		certProvider       string
+		vaultToken         string
+		caBundleSecretName string
+		issuerName         string
+		expectError        bool
+	}{
+		{
+			name:         "Cert Provider : Tresor",
+			certProvider: providers.TresorKind.String(),
+			expectError:  false,
+		},
+		{
+			name:         "Cert Provider : Vault and token is not empty",
+			certProvider: providers.VaultKind.String(),
+			vaultToken:   "anythinghere",
+			expectError:  false,
+		},
+		{
+			name:         "Cert Provider : Vault and token is empty",
+			certProvider: providers.VaultKind.String(),
+			vaultToken:   "",
+			expectError:  true,
+		},
+		{
+			name:               "Cert Provider : Cert-Manager with valid caBundleSecretName and certmanagerIssuerName",
+			certProvider:       providers.CertManagerKind.String(),
+			caBundleSecretName: "test-secret",
+			issuerName:         "test-issuer",
+			expectError:        false,
+		},
+		{
+			name:               "Cert Provider : Cert-Manager with valid caBundleSecretName and no certmanagerIssuerName",
+			certProvider:       providers.CertManagerKind.String(),
+			caBundleSecretName: "test-secret",
+			issuerName:         "",
+			expectError:        true,
+		},
+		{
+			name:               "Cert Provider : Cert-Manager with no caBundleSecretName and no certmanagerIssuerName",
+			certProvider:       providers.CertManagerKind.String(),
+			issuerName:         "",
+			caBundleSecretName: "",
+			expectError:        true,
+		},
+		{
+			name:         "Cert Provider : InvalidProvider",
+			certProvider: "InvalidProvider",
+			expectError:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
Let's move the `assert := tassert.New(t)` from the top of the test to here inside the sub-test so if a test case fails it's easier to pinpoint which sub-test the error message refers to.

(Also applicable to `TestValidateCLIParams` below)